### PR TITLE
[locator] show field where text has been found in current layer filter

### PR DESCRIPTION
**Pull request by 3nids**
also fix an issue that a field was not returning a match for big number (e/f formats)

### CODE REVIEW

File: src/app/locator/qgsinbuiltlocatorfilters.cpp [hunk 1]
```diff
@@ -234,7 +234,7 @@ void QgsActiveLayerFeaturesLocatorFilter::prepare( const QString &string, const
     }
     else if ( allowNumeric && field.isNumeric() )
     {
-      expressionParts << QStringLiteral( "%1 = %2" ).arg( QgsExpression::quotedColumnRef( field.name() ) ).arg( numericValue );
+      expressionParts << QStringLiteral( "%1 = %2" ).arg( QgsExpression::quotedColumnRef( field.name() ) ).arg( numericValue, 0, trunc( numericValue ) == numericValue ? 'f' : 'g' );
```
Comment (3nids) [inline src/app/locator/qgsinbuiltlocatorfilters.cpp]: not sure this is the best approach? shall we use `f` in any case?

**Comment by nyalldawson** [comment]
Looks good to me!

**Review by nyalldawson — APPROVED** [review]

File: src/app/locator/qgsinbuiltlocatorfilters.cpp (hunk 1 above)
Comment (nyalldawson) [reply src/app/locator/qgsinbuiltlocatorfilters.cpp]: I've no strong opinions either way. Go with your gut 👍

**Comment by 3nids** [comment]
now using string comparison instead of numerical. Seems to work better, and the same technique is actually used in attribute table filter.

**Comment by 3nids** [comment]
@nyalldawson  still good for you?

**Comment by 3nids** [comment]
(please squash if you merge)

**Review by nyalldawson — APPROVED** [review]

**Pull request merged**